src/ModuleResolver.ts

```diff
@@ -331,6 +332,20 @@ export class ModuleResolver implements Disposable {
           }
         }
 
+        if (fs.existsSync(path.join(dir, ".prettier-vscode-root"))) {
```

Comment:
I'm not so sure about this. What is the purpose? VS Code already defines what a root of the project is, why not just use that?

Reply:
What if you open not a project, but a subdirectory? You will have different root, so prettier can be resolved to a different version, it might be tricky for devs to debug such setup.

Reply:
This was actually introduced in my last PR (just copy pasted for the new code path of this PR). Here's more context on it: https://github.com/prettier/prettier-vscode/pull/1305#discussion_r405109478

TL;DR is that it was originally necessary for the tests, but could _technically_ have newfound usefulness in the scenario where you want to use an implicit dep instead of an explicit further up in the tree. I also agree with @PinkaminaDianePie that I'd be hesitant to depend on the VS Code root since it could break peoples workflows, esp in mono-repo environments.

Honestly I still don't think the flag would/should get usage outside of this project, which is why I elected to avoid documenting it. It's definitely a bit of a weird escape-hatch, so open to alternatives!

Reply:
I updated the code to strip this out in prod builds as it is for testing only: https://github.com/prettier/prettier-vscode/commit/1fe54f1134058cb684329dfa69ccda1716a07172

Reply:
Looks like that commit only handled one of two locations after this PR was also merged. Tossed up a PR to fix this and the failing mater tests: https://github.com/prettier/prettier-vscode/pull/1356